superset/assets/src/dashboard/reducers/getInitialState.js

```diff
@@ -92,6 +86,26 @@ export default function(bootstrapData) {
       };
 
       sliceIds.add(key);
+
+      // if chart is newly added from explore view,
+      // add a row in layout
+      if (!chartIdToLayoutId[key] && layout[parentId]) {
+        const parent = layout[parentId];
+        const rowContainer = getRowContainer();
+        layout[rowContainer.id] = rowContainer;
+        parent.children.push(rowContainer.id);
+
+        const chartHolder = getChartHolder({
+          size_x: 16,
+          size_y: 16,
+          slice_id: slice.slice_id,
+        });
+
+        layout[chartHolder.id] = chartHolder;
+        rowContainer.children.push(chartHolder.id);
+        chartIdToLayoutId[chartHolder.meta.chartId] = chartHolder.id;
+        hasUnsavedChanges = true;
```

Comment:
we could come back to this, but I think this may be confusing to users to see unsaved changes when they think they haven't done anything...seems like the main option would be to push the saved changes to the backend in that case but I'm not sure that's great either 🤔